src/orange/OrangeTrackView.hh

```diff
@@ -144,13 +142,21 @@ class OrangeTrackView
     // Create local distance
     inline CELER_FUNCTION detail::TempNextFace make_temp_next() const;
 
-    inline CELER_FUNCTION detail::LocalState make_local_state() const;
+    inline CELER_FUNCTION detail::LocalState
+    make_local_state(LevelId level) const;
 
     // Whether the next distance-to-boundary has been found
     CELER_FORCEINLINE_FUNCTION bool has_next_step() const;
 
     // Invalidate the next distance-to-boundary
     CELER_FORCEINLINE_FUNCTION void clear_next_step();
+
+    // Make a LevelStateAccessor for the current thread and level
+    CELER_FORCEINLINE_FUNCTION LevelStateAccessor make_lsa() const;
+
+    // Make a LevelStateAccessor for a given thread and level
```

Comment:
I think it only makes sense to do this for the current thread? the track view is *always* for a single thread.

Reply:
This is currently being used in the DetailedInitializer init function:

```    
        auto lsa = this->make_lsa(thread_, LevelId{i});
        auto lsa_other = this->make_lsa(init.other.thread_, LevelId{i});
 ```

I will remove it if we come up with a better way of doing this copy operation (which you comment on below)

Reply:
Actually, assuming I understand what you mean, I think we need still need to keep this second `make_lsa()` function unless you want me to construct an LSA ad hoc when it is used for a non-current thread (which I think is just one time).

Reply:
I *think* it should be true that `&this->state = &other.state` in general in the code, but that's a big assumption to make. You should be using `lsa_other = other->make_lsa(LeavelId{i})` for the second statement. There should never (at this point at least, since threads aern't cooperative) be any access across threads inside this level of the code.